pex/variables.py

```diff
@@ -68,40 +74,47 @@ def _get_kv(cls, variable):
             return kv
 
     def __init__(self, environ=None, rc=None, use_defaults=True):
+        # type: (Optional[Dict[str, str]], Optional[str], bool) -> None
         self._use_defaults = use_defaults
-        self._environ = (environ if environ is not None else os.environ).copy()
+        self._environ = environ.copy() if environ is not None else os.environ.copy()
         if not self.PEX_IGNORE_RCFILES:
             rc_values = self.from_rc(rc).copy()
             rc_values.update(self._environ)
             self._environ = rc_values
 
     def copy(self):
+        # type: () -> Dict[str, str]
         return self._environ.copy()
 
+    # TODO(#1041): Use a TypeVar once we require Python 3 to make this more precise.
     def _defaulted(self, default):
+        # type: (Optional[Union[str, int, bool]]) -> Optional[Union[str, int, bool]]
         return default if self._use_defaults else None
 
     def _get_bool(self, variable, default=False):
+        # type: (str, Optional[bool]) -> Optional[bool]
         value = self._environ.get(variable)
-        if value is not None:
-            if value.lower() in ("0", "false"):
-                return False
-            elif value.lower() in ("1", "true"):
-                return True
-            else:
-                die("Invalid value for %s, must be 0/1/false/true, got %r" % (variable, value))
-        else:
-            return self._defaulted(default)
+        if value is None:
+            return self._defaulted(default)  # type: ignore[return-value]
```

Comment:
We have to ignore due to the imprecise `._defaulted` method.

Reply:
Might this be the place for a `cast()`?

Reply:
We can't because it returns `Optional[bool]` :/

Reply:
That's quite annoying.

Reply:
Noted in #1042, but you can use cast, just pass the types as strings when those types (or aliases) are only available at `TYPE_CHECKING` time.